From: Chad Dupuis <cdupuis@redhat.com>
Date: Tue, 3 Aug 2010 13:49:58 -0400
Subject: [net] netxen: use DEFINE_PCI_DEVICE_TABLE()
Message-id: <20100803134713.2239.25386.sendpatchset@localhost.localdomain>
Patchwork-id: 27320
O-Subject: [RHEL 5.6 PATCH 18/44] drivers/net/: use DEFINE_PCI_DEVICE_TABLE()
Bugzilla: 562937
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>

Bugzilla
========

562937

Upstream Status
===============

net-2.6 commit id a3aa18842a5303fc28fcc4d57dbd16618bd830a0

Description
===========

>From 63dc115a6ece1fbde6354896f1bb3a3b65cf91ef Mon Sep 17 00:00:00 2001
From: Alexey Dobriyan <adobriyan@gmail.com>
Date: Thu, 7 Jan 2010 11:58:11 +0000
Subject: [PATCH] drivers/net/: use DEFINE_PCI_DEVICE_TABLE()

Use DEFINE_PCI_DEVICE_TABLE() so we get place PCI ids table into correct section
in every case.

Signed-off-by: Alexey Dobriyan <adobriyan@gmail.com>
Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/drivers/net/netxen/netxen_nic.h b/drivers/net/netxen/netxen_nic.h
index ad067c5..0e3c084 100644
--- a/drivers/net/netxen/netxen_nic.h
+++ b/drivers/net/netxen/netxen_nic.h
@@ -1433,4 +1433,9 @@ extern int netxen_rom_fast_read(struct netxen_adapter *adapter, int addr,
 
 extern struct ethtool_ops netxen_nic_ethtool_ops;
 
+#ifndef DEFINE_PCI_DEVICE_TABLE
+#define DEFINE_PCI_DEVICE_TABLE(_table) \
+         struct pci_device_id _table[] __devinitdata
+#endif
+
 #endif				/* __NETXEN_NIC_H_ */
diff --git a/drivers/net/netxen/netxen_nic_main.c b/drivers/net/netxen/netxen_nic_main.c
index a6a11ea..96a8a4a 100644
--- a/drivers/net/netxen/netxen_nic_main.c
+++ b/drivers/net/netxen/netxen_nic_main.c
@@ -105,7 +105,7 @@ static void netxen_config_indev_addr(struct net_device *dev, unsigned long);
 	{PCI_DEVICE(0x4040, (device)), \
 	.class = PCI_CLASS_NETWORK_ETHERNET << 8, .class_mask = ~0}
 
-static struct pci_device_id netxen_pci_tbl[] __devinitdata = {
+static DEFINE_PCI_DEVICE_TABLE(netxen_pci_tbl) = {
 	ENTRY(0x0001),
 	ENTRY(0x0002),
 	ENTRY(0x0003),
